src/Illuminate/Database/Eloquent/InvalidIdFormatException.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Support\Arr;
+
+/**
+ * @template TModel of \Illuminate\Database\Eloquent\Model
+ */
+class InvalidIdFormatException extends \RuntimeException
+{
+    /**
+     * Name of the affected Eloquent model.
+     *
+     * @var class-string<TModel>
+     */
+    protected $model;
+
+    /**
+     * The affected model IDs.
+     *
+     * @var array<int, int|string>
+     */
+    protected $ids;
+
+    /**
+     * Set the affected Eloquent model and instance id.
+     *
+     * @param  class-string<TModel>  $model
+     * @param  array<int, int|string>|int|string  $ids
+     * @param  string|null  $field
+     * @return $this
+     */
+    public function setModel($model, $ids = [], ?string $field = null)
+    {
+        $this->model = $model;
+        $this->ids = Arr::wrap($ids);
```

Comment:
If you type-hint `$ids`, you can simply do:
```suggestion
    public function setModel($model, array|int|string $ids = [], ?string $field = null)
    {
        $this->model = $model;
        $this->ids = (array) $ids;
```